### DISCUSSION TITLE
RNW + WinAppSDK Roadmap

### DISCUSSION THREAD

**Discussion by stmoy** [microsoft/react-native-windows#8906] [General]
🛠️ [**Update – May 2022**] 👋 Hello all! We have been diligently executing on the roadmap below and want to share some updates. The purpose of this discussion remains to:
1. 🔍 Provide clarity and transparency into how the platform team is thinking about WinAppSDK
2. 🗣️ Provide a space for you to provide feedback about this plan

As you might know, the new React Native rearchitecture (aka Fabric) has shipped as [“opt in” in version 0.68](https://reactnative.dev/blog/2022/03/30/version-068#opting-in-to-the-new-architecture), released in March. To help ensure that RNW continues to stay current with RN Core, we’re migrating our investments to align with the rearchitecture. In order to do that, we need to take a close look at the relationship between React Native core and the native Windows platform - and this investigation has taken us a bit longer than we expected. 

We’re taking this opportunity to align across the RN Core rearchitecture as well as the upcoming releases of WinAppSDK. As we continue this migration, we’ve identified some feature gaps in WinAppSDK that we’re working closely with the WinAppSDK team to address. 

**Our goal is still to have a version of React Native for Windows that combines the best parts of Fabric and WinAppSDK by the end of 2022.** We will provide more specific information as we get further along in this investigation. 

**Roadmap**
_Q1 2022_
- [RN 0.68 released](https://reactnative.dev/blog/2022/03/30/version-068)
- [WinAppSDK 1.0 released](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/)
- Technical exploration of options for RNW + WinAppSDK

_Q2 2022_
- [RN 0.69 release](https://github.com/facebook/react-native/releases)
- [WinAppSDK 1.1 release](https://github.com/microsoft/WindowsAppSDK#roadmap)

_Q3 2022_
- [RN 0.70 release](https://github.com/facebook/react-native/releases)
- Execute on plan for RNW + WinAppSDK

_Q4 2022_
- [WinAppSDK 1.2 release](https://github.com/microsoft/WindowsAppSDK#roadmap)
- Release RNW + WinAppSDK alongside the corresponding RN core version (0.71?)

### Open Issues/Questions

1. What are the specific "exit criteria" needed to flip the default mode of RNW from System Xaml + WinUI2 to WinAppSDK, particularly given the context around [UWP support in WinAppSDK](https://github.com/microsoft/WindowsAppSDK/discussions/1615)?
1. What are the steps for an existing RNW+WinUI2 app to migrate to a RNW+WinAppSDK app?
1. How many 3rd party RNW apps are developed for Xbox specifically?
1. How do we update the [existing library of community modules](https://reactnative.directory/?windows=true) that currently work with System Xaml + WinUI 2 to work with WinAppSDK? Is there a way to build a shim or abstraction layer?
1. Is it possible to align this roadmap with the [React Native re-architecture](https://reactnative.dev/blog/2021/08/19/h2-2021)?

**Comment by landongn** [comment]
Thanks a bunch for the clarity! This is extremely helpful.

**Comment by lukemcdo** [comment]
I don't think that switching from UWP to WinAppSDK by default makes sense until notifications are native and performant. If this can be achieved without official support from WinAppSDK then that is fine, but at the moment it's an extra piece of work in the transition to WinAppSDK.

**Comment by muramasa** [comment]
Hi @stmoy ,
Thanks for the roadmap! That is so helpful.

I want to clarify what is supporting with WinAppSDK.
RNW on WinAppSDK (UWP + WinUI3 ) or WinAppSDK (Desktop + WinUI3) or both?

**Reply by asklar** [reply]
hi @muramasa, we plan to support WinAppSDK stable releases, so Desktop + WinUI3.

**Reply by muramasa** [reply]
Hi @asklar ,
Thanks for the clarification! I’m excited about it.

**Reply by muramasa** [reply]
Hello again @asklar ,
I saw https://github.com/microsoft/react-native-windows/pull/9177 and I'm happy with that.
I have question about C++ project for WindowsAppSDK. https://github.com/microsoft/react-native-windows/issues/8132
Is that supported soon or no plan to support?

**Reply by asklar** [reply]
Yes, we plan to support C++ projects too; there is a lot we have to do in order to make WinAppSDK support be production ready so we will proceed in stages. We will likely have more details on when we expect certain features to land in the new year. For now you can see the work in the project board:  https://github.com/microsoft/react-native-windows/projects/30

**Reply by muramasa** [reply]
Hello @asklar again,

I saw https://github.com/microsoft/react-native-windows/pull/9177 PR.
I'm excited about that but I found some issue.

- I cannot create unpacked app https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-unpackaged-apps . Should it be supported?
- It doesn't follow window resize. It also doesn't render 2nd time (1st react tree is rendered. When I try to render replaced react tree, it isn't rendered. Update just text is working). I found some small issue in that (example: input field), too. Should I wait fix or contribute about it? (I'm not sure I can fix them by myself though)

**Reply by agniuks** [reply]
@muramasa 
Thanks for giving this a go already! We plan to support unpackaged apps as well and that will likely come around the same time as C++ support. 

Feel free to open issues in this repo for any bugs that you find, this will help us with tracking the work and will also help you know where we are at with putting in a fix. Here's the work that we'll tackling next: https://github.com/microsoft/react-native-windows/issues/9600

**Reply by muramasa** [reply]
@AgneLukoseviciute Sure, https://github.com/microsoft/react-native-windows/issues/9975 I created one issue for unpackging issue. I file the issue with C# but I'm facing the same issue in C++ project + unpackaged.

**Comment by dfoxg** [comment]
Are there any updates on this?
